Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp

```diff
@@ -958,4 +959,25 @@ void View::setCursor(const WebCore::Cursor& cursor)
 #endif
 }
 
+void View::callAfterNextPresentationUpdate(CompletionHandler<void()>&& callback)
+{
+#if ENABLE(WPE_PLATFORM)
+    if (m_wpeView) {
+        RELEASE_ASSERT(!m_nextPresentationUpdateCallback);
+        m_nextPresentationUpdateCallback = WTFMove(callback);
+        RELEASE_ASSERT(!m_bufferRenderedID);
+        m_bufferRenderedID = g_signal_connect_after(m_wpeView.get(), "buffer-rendered", G_CALLBACK(+[](WPEView* view, WPEBuffer*, gpointer userData) {
+            auto& webView = *reinterpret_cast<View*>(userData);
+            g_signal_handler_disconnect(view, std::exchange(webView.m_bufferRenderedID, 0));
+            webView.m_nextPresentationUpdateCallback();
+        }), this);
```

Comment:
AFAIU while GLib signals should be fast, setup/teardown can be slow due to looking signal information by their name, setting up marshalling, closures, etc. and there are locks involved.

Maybe we should connect this once on first use and then only check whether `webView.m_nextPresentationUpdateCallback` is non-null before use in the signal handler callback.

Reply:
I think signals are not fast? I would generally not emit a GObject signal in performance-sensitive code (unless you're confident the impact is not too bad).

Reply:
(The usual workaround, if it turns out to be too slow, would be to register a callback instead.)